Content.Server/GameObjects/Components/Projectiles/ProjectileComponent.cs

```diff
@@ -91,6 +98,8 @@ void ICollideBehavior.CollideWith(IEntity entity)
                 {
                     damage.ChangeDamage(damageType, amount, false, shooter);
                 }
+
+                _damagedEntity = true;
             }
```

Comment:
Is there any issue with deleting this entity upon collision?

Reply:
I am not entirely confident with how Dictionary works. If bullet is deleted upon collision, then in Robust.Shared.SharedPhysicsSystem when it iterates through the keys of collisionsWith dictionary at the end of ProcessCollisions method, would the dictionary still contains the key representing bullet's ICollideBehavior? Or would that key automatically get erased from the dictionary? Since if it doesn't then the behavior variable in the loop might be null (since bullet entity is already deleted) and there might be errors when behavior.PostCollide is called